utils/stall_free.hh

```diff
@@ -241,8 +250,17 @@ future<> clear_gently(T& c) noexcept {
     });
 }
 
-template <Container T>
+template <Extractable T>
 requires (!StringLike<T> && !Sequence<T> && !MapLike<T>)
+future<> clear_gently(T& c) noexcept {
+    return do_until([&c] { return c.empty(); }, [&c] {
+        auto node = c.extract(c.begin());
+        return internal::clear_gently(node.value()).finally([node = std::move(node)] {});
```

Comment:
unrelated:

The use of `internal` is unhappy, because we also have `seastar::internal` and can use it by accident.

Maybe it's time to remove `using namespace seastar` and instead import only select identifiers into the global namespace, like future and make_ready_future.